### ISSUE TITLE
Compilation issues

### ISSUE THREAD

**Issue by alasdairdc** [dimitri/pgloader#455]
TL;DR Building against CCL from source on `Debian 8.6 (Jessie) 64bit` requires multiarch and `freetds-dev:i386` even if you specify `CL=ccl64`.

Due to issues having with the current debian version e.g. https://github.com/dimitri/pgloader/issues/327 https://github.com/dimitri/pgloader/issues/453 decided to repcompile current master (https://github.com/dimitri/pgloader/commit/4a431498cabbac4095eb35b654eff43cd434a186) here is my experience / issues.

```
make clean && make pgloader
```

results in:

```
...
[package cl-postgres-error]................; 
; caught ERROR:
;   READ error during COMPILE-FILE:
;   
;     The symbol "HOST-ENT-ADDRESS-TYPE" is not external in the SB-BSD-SOCKETS package.
;   
;       Line: 99, Column: 73, File-Position: 4068
;   
;       Stream: #<SB-SYS:FD-STREAM
;                 for "file pgloader/build/quicklisp/dists/quicklisp/software/postmodern-20160929-git/cl-postgres/public.lisp"
;                 {10051C0683}>

debugger invoked on a UIOP/LISP-BUILD:COMPILE-FILE-ERROR in thread
```

Lets try the ccl version as that may solve my issues anyway (https://github.com/dimitri/pgloader/issues/327). SVN checked out ccl and copied / updated `scritps/ccl{,64}` to `/usr/local/bin` then:

```
make clean && make CL=ccl pgloader
```

results in (used the GO command to ignore mssql as will not be using it):

```
...
[package mssql]...
> Error: Unable to load foreign library (SYBDB).
>          Error opening shared library libsybdb.so : libsybdb.so: cannot open shared object file: No such file or directory.
> While executing: CFFI::FL-ERROR, in process listener(1).
> Type :GO to continue, :POP to abort, :R for a list of available restarts.
> If continued: Skip evaluation of (ql:quickload "pgloader")
> Type :? for other options.
1 > :GO
touch build/libs.stamp
ccl --no-init --load build/quicklisp/setup.lisp                \
             --eval '(ql:write-asdf-manifest-file "build/manifest.ql")'  \
             --eval '(quit)'
mkdir -p build/bin
ccl --no-init --load build/quicklisp/setup.lisp               \
             --eval '(ql:quickload "buildapp")'                   \
             --eval '(buildapp:build-buildapp "build/bin/buildapp.ccl")'              \
             --eval '(quit)'
To load "buildapp":
  Install 1 Quicklisp release:
    buildapp
; Fetching #<URL "http://beta.quicklisp.org/archive/buildapp/2015-12-18/buildapp-1.5.6.tgz">
; 16.00KB
==================================================
16,389 bytes in 0.00 seconds (16004.88KB/sec)
; Loading "buildapp"
[package buildapp]...........
;; loading system "buildapp"
mkdir -p build/bin
build/bin/buildapp.ccl      --logfile /tmp/build.log                \
                                                 \
                         --sbcl ccl                            \
                         --asdf-path .                           \
                         --asdf-tree build/quicklisp/local-projects     \
                         --manifest-file build/manifest.ql             \
                         --asdf-tree build/quicklisp/dists              \
                         --asdf-path .                           \
                         --load-system pgloader               \
                         --load src/hooks.lisp                   \
                         --entry pgloader:main                   \
                         --dynamic-space-size 4096         \
                                             \
                         --output build/bin/pgloader.tmp
;; loading system "pgloader"
Fatal MISSING-DEPENDENCY:
  Component #:UUID not found, required by #<SYSTEM "pgloader">
Fatal error in "buildapp" : Error #<SILENT-EXIT-ERROR #x1831B406>
Makefile:133: recipe for target 'build/bin/pgloader' failed
make: *** [build/bin/pgloader] Error 255
```

supplying the library path to `libsybdb.so`:

```
make clean && LD_LIBRARY_PATH=/usr/lib/x86_64-linux-gnu/ make CL=ccl pgloader
```

results in:

```
...
[package mssql]...
> Error: Unable to load foreign library (SYBDB).
>          Error opening shared library libsybdb.so : libsybdb.so: wrong ELF class: ELFCLASS64.
> While executing: CFFI::FL-ERROR, in process listener(1).
> Type :GO to continue, :POP to abort, :R for a list of available restarts.
> If continued: Skip evaluation of (ql:quickload "pgloader")
> Type :? for other options.
1 > 
```

Seems it is trying to build 32-bit version of code.and `:GO` still fails with:

```
...
Fatal MISSING-DEPENDENCY:
  Component #:UUID not found, required by #<SYSTEM "pgloader">
```

Okay so lets try using CCL64:

```
make clean && make CL=ccl64 pgloader
```

results in:

```
...
build/bin/buildapp.ccl      --logfile /tmp/build.log                \
                                                 \
                         --sbcl ccl64                            \
                         --asdf-path .                           \
                         --asdf-tree build/quicklisp/local-projects     \
                         --manifest-file build/manifest.ql             \
                         --asdf-tree build/quicklisp/dists              \
                         --asdf-path .                           \
                         --load-system pgloader               \
                         --load src/hooks.lisp                   \
                         --entry pgloader:main                   \
                         --dynamic-space-size 4096         \
                                             \
                         --output build/bin/pgloader.tmp
;; loading system "pgloader"
Fatal SIMPLE-ERROR:
  Compilation failed: In (SOCKET-SHUTDOWN (STREAM-USOCKET T)) inside an anonymous lambda form: Undeclared free variable SOCK in pgloader/build/quicklisp/dists/quicklisp/software/usocket-0.6.4.1/backend/openmcl.lisp
Fatal error in "buildapp" : Error #<SILENT-EXIT-ERROR #x1831631E>
Makefile:133: recipe for target 'build/bin/pgloader' failed
make: *** [build/bin/pgloader] Error 255
```

hmmm, still failing.

Installing `freetds-dev:i386` and running:

```
make clean && make CL=ccl pgloader
```

results in:

```
...
mv build/bin/pgloader.tmp build/bin/pgloader
```

so was successful - using 32-bit (multiarch) freetds-dev

Software versions:
- Debain 8.6 Jessie 64bit
- pgloader current master (https://github.com/dimitri/pgloader/commit/4a431498cabbac4095eb35b654eff43cd434a186)
- sbcl 1.2.4-2 (from main debian repo)
- unzip 6.0-16+deb8u2 (from main debian repo)
- libsqlite3-dev 3.8.7.1-1+deb8u2 (from main debian repo)
- make 4.0-8.1 (from main debian repo)
- curl 7.38.0-4+deb8u4 (from main debian repo)
- gawk 4.1.1+dfsg-1 (from main debian repo)
- freetds-dev 0.91-6+b1 (from main debian repo)
- freetds-dev:i386 0.91-6+b1 (from main debian repo)
- libzip-dev 0.11.2-1.2 (from main debian repo)
- ccl SVN revision 16796 (from http://svn.clozure.com/publicsvn/openmcl/release/1.11/linuxx86/ccl)

**Comment by dimitri** [comment]
Thanks for the detailed report!

**Comment by Ahuri3** [comment]
Any solution if you wish to build against sbcl ? I'm currently having the same problems mentioned above but CCL might not be an option in my case.

**Comment by alasdairdc** [comment]
@Ahuri3 You'd be better checking to see if someone has experienced your same issue (or create a new one).
